src/node.cc

```diff
@@ -1269,41 +1264,22 @@ Local<Value> MakeCallback(Environment* env,
     }
   }
 
-  if (ran_init_callback && !pre_fn.IsEmpty()) {
-    TryCatch try_catch(env->isolate());
-    MaybeLocal<Value> ar = pre_fn->Call(env->context(), object, 0, nullptr);
-    if (ar.IsEmpty()) {
-      ClearFatalExceptionHandlers(env);
-      FatalException(env->isolate(), try_catch);
-      return Local<Value>();
-    }
-  }
+  // TODO(trevnorris): Correct this once node::MakeCallback() support id and
+  // triggerId. Consider completely removing it until then so the async id can
+  // propagate through to the fatalException after hook calls.
```

Comment:
It's currently missing from `node::MakeCallback()` b/c I'm not sure how to deal with it. I could add an API where the optional `asyncId` is passed to `node::MakeCallback()`, but it's not guaranteed the users will pass it in. I also can't rely on a specific object property b/c the user can pass in whatever they want. The coming native API will force this, but I'm not sure of a fix for the existing API.